plugin/trino-hive/src/main/java/io/trino/plugin/hive/projection/DateProjection.java

```diff
@@ -224,17 +224,27 @@ private static Supplier<Instant> parseDateRangerBound(String columnName, String
                 Instant now = Instant.now();
                 return () -> now;
             }
-            throw invalidRangeProperty(columnName, dateFormat.toPattern(), Optional.of("Invalid expression"));
+            throw invalidRangeProperty(columnName, dateFormatPattern, Optional.of("Invalid expression"));
         }
 
-        Instant dateBound;
-        try {
-            dateBound = dateFormat.parse(value).toInstant();
-        }
-        catch (ParseException e) {
-            throw invalidRangeProperty(columnName, dateFormat.toPattern(), Optional.of(e.getMessage()));
+        return () -> {
+            try {
+                return parse(value, dateFormat);
+            }
+            catch (DateTimeParseException e) {
+                throw invalidRangeProperty(columnName, dateFormatPattern, Optional.of(e.getMessage()));
+            }
+        };
+    }
+
+    private static Instant parse(String value, DateTimeFormatter dateFormat)
```

Comment:
I would inline this into the usage site since there's only one caller and the wrapping try / catch context is important.

Reply:
ah, it's actually referenced in two places: `parseDateRangeBound` and `adjustBoundToDateFormat`. The error messages are slightly different.